solana/solana-ibc/programs/solana-ibc/src/transfer/mod.rs

```diff
@@ -142,12 +144,112 @@ impl ibc::Module for IbcStorage<'_, '_> {
             .into_bytes(),
             ..packet.clone()
         };
-        let (extras, ack) = ibc::apps::transfer::module::on_recv_packet_execute(
-            self,
-            &maybe_ft_packet,
-        );
-        let ack_status = str::from_utf8(ack.as_bytes())
+        let (extras, mut ack) =
+            ibc::apps::transfer::module::on_recv_packet_execute(
+                self,
+                &maybe_ft_packet,
+            );
+        let cloned_ack = ack.clone();
+        let ack_status = str::from_utf8(cloned_ack.as_bytes())
             .expect("Invalid acknowledgement string");
+        let status = serde_json::from_slice::<ibc::AcknowledgementStatus>(
+            ack.as_bytes(),
+        );
+        let success = if let Ok(status) = status {
+            status.is_successful()
+        } else {
+            ack = ibc::AcknowledgementStatus::error(
+                ibc::TokenTransferError::AckDeserialization.into(),
+            )
+            .into();
+            false
+        };
+        fn call_bridge_escrow(
+            accounts: &[AccountInfo],
+            data: Vec<u8>,
+        ) -> Result<(), ibc::AcknowledgementStatus> {
+            // Perform hooks
+            let data = match serde_json::from_slice::<PacketData>(&data) {
+                Ok(data) => data,
+                Err(_) => {
+                    return Err(ibc::AcknowledgementStatus::error(
+                        ibc::TokenTransferError::PacketDataDeserialization
+                            .into(),
+                    ));
+                }
+            };
+            // The hook would only be called if the transferred token is the one we are interested in
+            if data.token.denom.base_denom.as_str() == HOOK_TOKEN_ADDRESS {
```

Comment:
```suggestion
            if data.token.denom.base_denom.as_str() != HOOK_TOKEN_ADDRESS {
                return Ok(());
            }
```

and now you can dedent the rest of the function.

Reply:
oh i see, yes this looks clean.